### ISSUE TITLE
[feature request] The separation of scheduler and cdnmgr

### ISSUE THREAD

**Issue by xq2248** [dragonflyoss/Dragonfly#437]
Supernode has two responsibilities: scheduler and cdnmgr. This two responsibilities may be separated into two different components.

Scheduler process the peer requests and consumes the CPU resources, but cdnmgr process the content and consumes both CPU and IO resources. The resource requirement of scheduler relies on the concurrency of peers which are scheduled by the same supernode. The resource requirement of cdnmgr mainly relies on the number of contents which are required to be distributed at the same time. So, the scale in/out policy of this two components are different. Through separation, cdnmgr can be deployed to other different devices, which can accelerate file downloading and clipping with more resources. High performance devices will not be necessary. Sometimes we are only servers with low performance to deploy supernodes, or we may deploy supernodes in existing devices with limited CPU and IO resources. 

The side effect is that, one scheduler and many cdnmgrs communicates through network API interfaces, the architecture will be more complicated.

**Comment by jyizheng** [comment]
Is this issue still open? I'd like to work on it.

**Comment by allencloud** [comment]
> Is this issue still open? I'd like to work on it.

Yes, still open. We are very glad that you could help to make it.
Maybe we can make a detailed design below this issue, review that by community and implement that via PR. @jyizheng
